src/main/webapp/dev/js/common/questionMsq.js

```diff
@@ -177,19 +177,19 @@ function toggleMsqMinSelectableChoices(questionNum) {
 }
 
 function toggleMsqGeneratedOptions(checkbox, questionNum) {
-    if (checkbox.checked) {
+    if ($(checkbox).prop('checked')) {
         $(`#msqChoiceTable-${questionNum}`).find('input[type=text]').prop('disabled', true);
         $(`#msqChoiceTable-${questionNum}`).hide();
         $(`#msqGenerateForSelect-${questionNum}`).prop('disabled', false);
         $(`#msqOtherOptionFlag-${questionNum}`).closest('.checkbox').hide();
-        $(`#generatedOptions-${questionNum}`).attr('value',
+        $(`#msqGeneratedOptions-${questionNum}`).attr('value',
                 $(`#msqGenerateForSelect-${questionNum}`).prop('value'));
     } else {
         $(`#msqChoiceTable-${questionNum}`).find('input[type=text]').prop('disabled', false);
         $(`#msqChoiceTable-${questionNum}`).show();
         $(`#msqGenerateForSelect-${questionNum}`).prop('disabled', true);
         $(`#msqOtherOptionFlag-${questionNum}`).closest('.checkbox').show();
-        $(`#generatedOptions-${questionNum}`).attr('value', 'NONE');
+        $(`#msqGeneratedOptions-${questionNum}`).attr('value', 'NONE');
```

Comment:
You might want to use `.val()` instead? Checkout this discussion https://stackoverflow.com/questions/8312820/jquery-val-vs-attrvalue

Most of the codebase uses .val() too anyway, would be good to be consistent